app/models/batch/tecan_behaviour.rb

```diff
@@ -17,14 +20,15 @@ def generate_tecan_data(target_barcode, override_plate_type = nil)
       full_source_barcode = request.asset.plate.machine_barcode
       full_destination_barcode = request.target_asset.plate.machine_barcode
 
-      source_plate_name = override_plate_type.presence || request.asset.plate.plate_type.tr('_', "\s")
-
       if data_object['source'][full_source_barcode].nil?
-        data_object['source'][full_source_barcode] = { 'name' => source_plate_name, 'plate_size' => request.asset.plate.size }
+        source_plate_type = override_plate_type.presence || request.asset.plate.plate_type.tr('_', "\s")
+        data_object['source'][full_source_barcode] = { 'name' => source_plate_type, 'plate_size' => request.asset.plate.size }
       end
+
       if data_object['destination'][full_destination_barcode].nil?
+        target_plate_type = (target_plate.plate_type || PlateType.cherrypickable_default_type).tr('_', "\s")
```

Comment:
Pull out the plate type for the destination in a similar way to the source. This uses the plate type which we recorded earlier. I've left the fallback to the default in cases where a type has not been set.